### DISCUSSION TITLE
Permission Manager (custom)

### DISCUSSION THREAD

**Discussion by Rudxain** [RikkaApps/Shizuku#320] [Ideas]
Shizuku should have a permission system, to allow users to have granular control over the ADB cmds that the client apps can execute.

For example, "App Ops" only needs to run `adb shell appops *` (and other cmds), it doesn't need `adb shell wm *`

**Comment by RikkaW** [comment]
Most Android APIs are only protected by Android permissions, the same UID has the same Android permissions. Shizuku apps run their codes under the same UID as the Shizuku server (if Shizuku is started by adb, the UID is 2000; for root, is 0).

However, it is impossible to drop some Android permissions at runtime. So, your requirement is impossible.

Btw, most Shizuku apps in the real world are not implemented by executing commands :)

**Reply by Rudxain** [reply]
I didn't explain correctly. I meant the app should have its own custom permissions that control API access (which are equivalent to commands, at a lower level). The app would have to remember the permissions itself, not the Android OS
